workspaces/marketplace/plugins/marketplace-backend/src/services/MarketplaceCatalogService.ts

```diff
@@ -83,4 +85,69 @@ export class MarketplaceCatalogService implements MarketplaceService {
 
     return result.items as MarketplacePluginList[];
   }
+
+  async getPluginListByName(name: string): Promise<MarketplacePluginList> {
+    const token = await this.auth.getPluginRequestToken({
+      onBehalfOf: await this.auth.getOwnServiceCredentials(),
+      targetPluginId: 'catalog',
+    });
+    const result = await this.catalog.getEntities(
+      {
+        filter: {
+          kind: 'pluginList',
+          'metadata.name': name,
+        },
+      },
+      token,
+    );
+
+    return result.items?.[0] as MarketplacePluginList;
+  }
+
+  async getPluginByName(name: string): Promise<MarketplacePluginEntry> {
+    const token = await this.auth.getPluginRequestToken({
+      onBehalfOf: await this.auth.getOwnServiceCredentials(),
+      targetPluginId: 'catalog',
+    });
+    const result = await this.catalog.getEntities(
+      {
+        filter: {
+          kind: 'plugin',
+          'metadata.name': name,
+        },
+      },
+      token,
+    );
+
+    return result.items?.[0] as MarketplacePluginEntry;
+  }
+
+  async getPluginsByPluginsListName(
+    name: string,
+  ): Promise<MarketplacePluginEntry[]> {
+    const pluginList = await this.getPluginListByName(name);
+    const plugins = pluginList?.spec?.plugins;
+
+    if (!pluginList) {
+      throw new NotFoundError(
+        `${MarketplaceKinds.pluginList}:${name} not found`,
+      );
+    }
```

Comment:
I guess getPluginListByName will already throw an error if it's not found?

```suggestion
```

Reply:
No, `getPluginListByName` will not throw error from this file. Error will be thrown by the consumer, in this case it is router.ts.